x-pack/plugins/saved_objects_tagging/public/management/tag_management_page.tsx

```diff
@@ -160,13 +178,51 @@ export const TagManagementPage: FC<TagManagementPageParams> = ({
     [overlays, notifications, fetchTags, tagClient]
   );
 
+  const executeBulkAction = useCallback(
+    async (action: TagBulkAction) => {
+      try {
+        await action.execute(selectedTags.map(({ id }) => id));
+        if (action.refreshAfterExecute) {
+          await fetchTags();
+        }
```

Comment:
Given that bulk delete could fail part of the way through the list of tags, I think it would make sense to also refresh the tags after a failure, not just in the success case.